docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+ "Prateek"
+ "Anam"
+
+julia> data[:, :City] .= ["Kanpur", "Lucknow", "Bhuvneshwar", "Jaipur", "Ranchi", "Dehradoon"] # allocates the column in-place because `:City` is not present in `data`
+6-element Vector{String}:
+ "Kanpur"
+ "Lucknow"
+ "Bhuvneshwar"
+ "Jaipur"
+ "Ranchi"
+ "Dehradoon"
+
+julia> data
+6×5 DataFrame
+ Row │ marks  subject           semester  Students  City
+     │ Int64  String            Int64     String    String
+─────┼──────────────────────────────────────────────────────────
+   1 │    85  English                  4  Rohit     Kanpur
+   2 │    89  Computer Science         5  Akshat    Lucknow
+   3 │    78  Economics                7  Rahul     Bhuvneshwar
+   4 │    58  History                  8  Aayush    Jaipur
+   5 │    96  Economics                2  Prateek   Ranchi
+   6 │    68  Biology                  1  Anam      Dehradoon
+```
+
+Note that if `:col` is not present in `data` then using `!` and `:` are equivalent. The major difference between in-place 
+and replace operations is that replacing columns is needed if new values have a different type than the old ones. For 
+instance here `!` works and `:` fails:
+
+```jldoctest dataframe
+julia> data
+6×5 DataFrame
+ Row │ marks  subject           semester  Students  City
+     │ Int64  String            Int64     String    String
+─────┼──────────────────────────────────────────────────────────
+   1 │    85  English                  4  Rohit     Kanpur
+   2 │    89  Computer Science         5  Akshat    Lucknow
+   3 │    78  Economics                7  Rahul     Bhuvneshwar
+   4 │    58  History                  8  Aayush    Jaipur
+   5 │    96  Economics                2  Prateek   Ranchi
+   6 │    68  Biology                  1  Anam      Dehradoon
+
+julia> data[:, :marks] .= "Economics"
+ERROR: MethodError: Cannot `convert` an object of type String to an object of type Int64
+
+julia> data[!, :marks] .= "Economics"
+6-element Vector{String}:
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+
+julia> data
+6×5 DataFrame
+ Row │ marks      subject           semester  Students  City
+     │ String     String            Int64     String    String
+─────┼──────────────────────────────────────────────────────────────
+   1 │ Economics  English                  4  Rohit     Kanpur
+   2 │ Economics  Computer Science         5  Akshat    Lucknow
+   3 │ Economics  Economics                7  Rahul     Bhuvneshwar
+   4 │ Economics  History                  8  Aayush    Jaipur
+   5 │ Economics  Economics                2  Prateek   Ranchi
+   6 │ Economics  Biology                  1  Anam      Dehradoon
+```
+
+In mostly cases, above as you can see for getting a column or assigning to a column instead of `data[!, :col]` 
+and `data[!, :col] = val` it is usually better to just write `data.col` and `data.col = val` respectively as 
+it is the same and simpler to type and read.
+
+## Not, Between, Cols, and All selectors
```

Comment:
why do you cover these selectors after `setindex!`. It seems that the content is not in a correct order, but maybe there is a reason for this?

Reply:
Again, our purpose is to explain every function and how to perform operations using those functions. So, according to me this order and place were fine.

Reply:
> Again, our purpose is to explain every function and how to perform operations using those functions

Agreed

> So, according to me this order and place were fine.

I am not sure I follow. Your current order is:
* `getindex`
* `setindex!`
* back to `getindex`

Why not explain `getindex` first for all cases and then switch to `setindex!`.

Reply:
As I have already discussed `getindex` and `setindex!`. So now it will give a clear insight to the users that what are these and then if for any other operations like, *selectors* I use it then I guess user won't be confused because I have already told them that what are these. So according to my opinion, it is fine. Thank you!